runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java

```diff
@@ -39,38 +38,28 @@
 	private final Courses course;
 	private Instant lastLapCompleted;
 	private int totalLaps;
-	private int lapsTillLevel;
 	private int lapsTillGoal;
 
 	AgilitySession(Courses course)
 	{
 		this.course = course;
 	}
 
-	void incrementLapCount(Client client)
+	void incrementLapCount(Client client, XpTrackerService xpTrackerService)
 	{
 		lastLapCompleted = Instant.now();
 		++totalLaps;
 
-		int currentExp = client.getSkillExperience(Skill.AGILITY);
-		int nextLevel = client.getRealSkillLevel(Skill.AGILITY) + 1;
+		final int currentExp = client.getSkillExperience(Skill.AGILITY);
+		final int goalXp = xpTrackerService.getEndGoalXp(Skill.AGILITY);
+		final int goalRemainingXp = goalXp - currentExp;
 
-		int remainingXp;
```

Comment:
I recall this being here for a specific reason: https://github.com/runelite/runelite/pull/2370#discussion_r186268067. Maybe it doesn't matter anymore since it isn't looking at the real skill level? Can someone test that?

Reply:
I'm not near a computer I can test with, but I recall the original code this was written for. Since goal XP doesn't have to be calculated (since XpTrackerService does that), a local variable is no longer needed to determine that.